File: MyApplication/app/src/main/java/io/ckl/boilerplate/modules/main/MainInterfaces.java
package io.ckl.boilerplate.modules.main;

import io.ckl.boilerplate.modules.base.BaseInterfaces;

public class MainInterfaces {

    public interface Presenter extends BaseInterfaces.Presenter {
        void onGreetButtonPressed();
    }

    public interface View extends BaseInterfaces.View {
        void showGreeting(String greeting);
    }

}

File: MyApplication/app/src/main/java/io/ckl/boilerplate/modules/main/MainInterfaces.kt
package io.ckl.boilerplate.modules.main

import io.ckl.boilerplate.modules.base.BaseInterfaces

class MainInterfaces {

    interface Presenter: BaseInterfaces.Presenter {
        fun onGreetButtonPressed()
    }

    interface View: BaseInterfaces.View {
        fun showGreeting(greeting: String)
    }

}
